subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -486,21 +537,33 @@ static int z_ztest_run_test_suite_ptr(struct ztest_suite_node *suite)
 	if (suite->setup != NULL) {
 		data = suite->setup();
 	}
-	while ((test = ztest_get_next_test(suite->name, test)) != NULL) {
-		fail += run_test(suite, test, data);
 
-		if (fail && FAIL_FAST) {
-			break;
+	for (int i = 0; i < NUM_ITER_PER_TEST; i++) {
+		fail = 0;
+
+#if (IS_ENABLED(CONFIG_ZTEST_DO_THE_SHUFFLE))
+		z_ztest_shuffle(_ztest_unit_test_list_start,
+				_ztest_unit_test_list_end - _ztest_unit_test_list_start, &tmp,
+				sizeof(struct ztest_unit_test));
+#endif
```

Comment:
If you remove the above `if` and always have the static function there, you can instead use `IS_ENABLED` like:

```
if (IS_ENABLED(CONFIG_ZTEST_DO_THE_SHUFFLE)) {
  z_ztest_shuffle(...);
}
```